electrum/gui/qt/password_dialog.py

```diff
@@ -166,55 +168,26 @@ def clear_password_fields(self):
             field.clear()
 
 
-class PasswordLayoutForHW(object):
-
-    def __init__(self, msg, wallet=None):
-        self.wallet = wallet
-
-        vbox = QVBoxLayout()
-        label = QLabel(msg + "\n")
-        label.setWordWrap(True)
-
-        grid = QGridLayout()
-        grid.setSpacing(8)
-        grid.setColumnMinimumWidth(0, 150)
-        grid.setColumnMinimumWidth(1, 100)
-        grid.setColumnStretch(1,1)
-
-        logo_grid = QGridLayout()
-        logo_grid.setSpacing(8)
-        logo_grid.setColumnMinimumWidth(0, 70)
-        logo_grid.setColumnStretch(1,1)
-
-        logo = QLabel()
-        logo.setAlignment(Qt.AlignmentFlag.AlignCenter)
-
-        logo_grid.addWidget(logo,  0, 0)
-        logo_grid.addWidget(label, 0, 1, 1, 2)
-        vbox.addLayout(logo_grid)
-
-        if wallet and wallet.has_storage_encryption():
-            lockfile = "lock.png"
-        else:
-            lockfile = "unlock.png"
-        logo.setPixmap(QPixmap(icon_path(lockfile))
-                       .scaledToWidth(36, mode=Qt.TransformationMode.SmoothTransformation))
-
-        vbox.addLayout(grid)
+class PasswordLayoutForHW(PasswordLayout):
 
-        self.encrypt_cb = QCheckBox(_('Encrypt wallet file'))
-        grid.addWidget(self.encrypt_cb, 1, 0, 1, 2)
-
-        self.vbox = vbox
-
-    def should_encrypt_storage(self):
+    def __init__(self, msg, kind, OK_button, wallet=None):
+        PasswordLayout.__init__(self, msg, kind, OK_button, wallet=wallet)
+        self.encrypt_cb = QCheckBox(_('Encrypt wallet file using hardware wallet device'))
+        self.encrypt_cb.setToolTip(_('Note: If you enable this setting, you will need your hardware device to open your wallet.'))
+        self.encrypt_cb.stateChanged.connect(self.on_encrypt_cb)
+        self.grid.addWidget(self.encrypt_cb, 4, 0, 1, 2)
+        self.encrypt_cb.setChecked(wallet.storage.is_encrypted_with_hw_device())
```

Comment:
we need type checks.
```suggestion
        self.encrypt_cb.setChecked(wallet.storage.is_encrypted_with_hw_device() if wallet else True)
```